### PR TITLE
Fixed some failing tests and marked some active issues

### PR DESCRIPTION
**Pull request by sepidehMS**
Some failing tests were fixed. Some tests which did not fail, but printed out error messages were either fixed or marked as active issues.

Relevant issue: 641
Newly created issues: 4054, 4057, 4063

### CODE REVIEW

**Comment by ellismg** [comment]
@sepidehMS I think you need to configure your git client username and e-mail.  Right now it is showing as "unknown".

git config --global user.name and user.email is what you want, if you use the command line client.  Then do a git commit --amend --reset-author.

File: src/NuGet.config [hunk 1]
```diff
@@ -1,16 +1,15 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <!-- NOTE: Leave this file here and keep it in sync with list in dir.props. -->
   <!-- The command-line doesn't need it, but the IDE does.                    -->
+  <config>
+    <add key="repositoryPath" value="..\packages" />
+  </config>
```
Comment (stephentoub) [inline src/NuGet.config]: Why is this file being updated?

File: src/System.Xml.XDocument/tests/axes/project.lock.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "locked": true,
+  "locked": false,
```
Comment (stephentoub) [inline src/System.Xml.XDocument/tests/axes/project.lock.json]: This should remain true.

**Comment by mellinoe** [comment]
To lock the project.lock.json files, you will need to run msbuild /p:LockDependencies=true when you build, and then the flag at the top should say "true"

**Comment by ellismg** [comment]
I don't understand why the lock file is even being updated at all.  Where is the corresponding change to a project.json file?

I would revert that change and rebase this branch on-top of master and see if building again causes it to get updated.

File: src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs [hunk 3]
```diff
@@ -148,7 +148,6 @@ public bool CompareReader(XDocument doc, string expectedXml)
             rs.DtdProcessing = DtdProcessing.Ignore;
             rs.CloseInput = true;
             _diff.Option = XmlDiffOption.IgnoreAttributeOrder;
-
```
Comment (sepidehMS) [inline src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs]: remove

**Comment by krwq** [comment]
Looks good :+1:  Thanks for investigating this!

**Comment by stephentoub** [comment]
FYI, several failing tests:
```
CoreXml.Test.XLinq.FunctionalTests+XNodeReaderTests+TCReadEndElement.TestReadEndElementOnEndElementWithoutNamespace [FAIL]
        System.Xml.XmlException : Data at the root level is invalid. Line 1, position 1.
        Stack Trace:
              at System.Xml.XmlTextReaderImpl.Throw(Exception e)
              at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
              at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
              at System.Xml.XmlTextReaderImpl.Read()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XDocument.cs(352,0): at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\XDocument.Common\BridgeHelpers.cs(83,0): at CoreXml.Test.XLinq.BridgeHelpers.GetReader()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\xNodeReader\CXmlReaderReadEtc.cs(848,0): at CoreXml.Test.XLinq.FunctionalTests.XNodeReaderTests.TCReadEndElement.TestReadEndElementOnEndElementWithoutNamespace()
     CoreXml.Test.XLinq.FunctionalTests+XNodeReaderTests+TCReadOuterXml.MovingToNonExistingAttributeThrowsAndDoesntBreakReaderState [FAIL]
        System.Xml.XmlException : '<', hexadecimal value 0x3C, is an invalid attribute character. Line 115, position 1.
        Stack Trace:
              at System.Xml.XmlTextReaderImpl.Throw(Exception e)
              at System.Xml.XmlTextReaderImpl.ParseAttributeValueSlow(Int32 curPos, Char quoteChar, NodeData attr)
              at System.Xml.XmlTextReaderImpl.ParseAttributes()
              at System.Xml.XmlTextReaderImpl.ParseElement()
              at System.Xml.XmlTextReaderImpl.ParseElementContent()
              at System.Xml.XmlTextReaderImpl.Read()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XContainer.cs(912,0): at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XContainer.cs(919,0): at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r, LoadOptions o)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XDocument.cs(374,0): at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\XDocument.Common\BridgeHelpers.cs(83,0): at CoreXml.Test.XLinq.BridgeHelpers.GetReader()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\xNodeReader\ReadOuterXml.cs(213,0): at CoreXml.Test.XLinq.FunctionalTests.XNodeReaderTests.TCReadOuterXml.MovingToNonExistingAttributeThrowsAndDoesntBreakReaderState()
     CoreXml.Test.XLinq.FunctionalTests+XNodeReaderTests+TCReadEndElement.TestReadEndElementOnEndElementWithNamespace [FAIL]
        System.Xml.XmlException : Unexpected end of file while parsing Name has occurred. Line 168, position 70.
        Stack Trace:
              at System.Xml.XmlTextReaderImpl.Throw(Exception e)
              at System.Xml.XmlTextReaderImpl.ParseQName(Boolean isQName, Int32 startOffset, Int32& colonPos)
              at System.Xml.XmlTextReaderImpl.ThrowTagMismatch(NodeData startTag)
              at System.Xml.XmlTextReaderImpl.ParseEndElement()
              at System.Xml.XmlTextReaderImpl.ParseElementContent()
              at System.Xml.XmlTextReaderImpl.Read()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XContainer.cs(912,0): at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XContainer.cs(919,0): at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r, LoadOptions o)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\src\System\Xml\Linq\XDocument.cs(374,0): at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\XDocument.Common\BridgeHelpers.cs(83,0): at CoreXml.Test.XLinq.BridgeHelpers.GetReader()
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Xml.XDocument\tests\xNodeReader\CXmlReaderReadEtc.cs(859,0): at CoreXml.Test.XLinq.FunctionalTests.XNodeReaderTests.TCReadEndElement.TestReadEndElementOnEndElementWithNamespace()
  Finished:    System.Xml.XDocument.xNodeReader.Tests
```

**Comment by stephentoub** [comment]
@sepidehMS, what's the status of this PR?

**Comment by sepidehMS** [comment]
@stephentoub Looks like there is a race condition between the test. Still investigating the issue.

**Comment by stephentoub** [comment]
@sepidehMS, what's the status of this PR?

**Comment by krwq** [comment]
This PR is substituted by https://github.com/dotnet/corefx/pull/5392

**Pull request closed without merging**